### PR TITLE
safely get the user input for their subdomain

### PR DESCRIPTION
**Pull request by James-Sangalli**

### CODE REVIEW

File: examples/ENS/subdomainBundle.en.js [hunk 1]
```diff
@@ -11,8 +11,9 @@ class Token {
         let ensName = this.props.ensName;
         let subdomainElement = document.getElementById("subdomain");
         subdomainElement.addEventListener("keyup", function() {
-            let subdomain = subdomainElement.value.toString().trim().toLowerCase().replace(ensName, "");
-            document.getElementById("subdomainHash").innerHTML = namehash.hash(subdomain + "." + ensName);
+            let subdomain = subdomainElement.value.toString().trim().toLowerCase();
+            document.getElementById("subdomainHash").innerHTML =
+                namehash.hash(subdomain.substring(0, subdomain.indexOf(".")) + "." + ensName);
```
Comment (hboon) [inline examples/ENS/subdomainBundle.en.js:16]: This line doesn't work correctly if `subdomain` doesn't contain a ".". Maybe be more strict about the value entered subdomain?

There are just 2 (or 3) possible types:

A) "name.${ensName}"
B) "name"
C) can "name" contain "."? eg. "my.sub.joe.eth"?

We can decide if we only want to support either one or all of them, and it'll be easier to code.

File: examples/ENS/subdomainBundle.en.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
-//<![CDATA[
 (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
+//<![CDATA[
```
Comment (hboon) [inline examples/ENS/subdomainBundle.en.js:2]: I think we should use `//<![CATA[` to wrap the entire file, or not at all. Either we escape all the `<`, etc or we don't, right?

File: examples/ENS/subdomainBundle.en.js (hunk 1 above)
Comment (James-Sangalli) [reply examples/ENS/subdomainBundle.en.js:16]: Support both

Comment (hboon) [reply examples/ENS/subdomainBundle.en.js]: Then how about a more specific if-else [1]. How about:

```
if subdomain.suffix == ".${ensName}" {
  //substring with the length of ".${ensName}"
} else {
  //depending on whether (C) above is possible and act accordingly
}
```

[1] `indexOf()` returns -1 if it's not found

Comment (hboon) [reply examples/ENS/subdomainBundle.en.js]: Alternative we can cheat a little and display the text box like:

Subdomain: [          ].joe.eth

And expect the user to never type (A)

Comment (James-Sangalli) [reply examples/ENS/subdomainBundle.en.js]: Hmm, that might be a bit more friendly

**Pull request merged**